tests/uptane_test.cc

```diff
@@ -296,8 +293,8 @@ TEST(Uptane, PetNameCreation) {
     Uptane::Repository uptane(conf, storage, http);
     EXPECT_TRUE(uptane.initialize());
 
-    EXPECT_TRUE(boost::filesystem::exists(device_path));
-    test_name2 = Utils::readFile(device_path);
+    EXPECT_TRUE(boost::filesystem::exists(temp_dir.Path() / "device_id"));
+    test_name2 = Utils::readFile(temp_dir.Path() / "device_id");
```

Comment:
It took me a minute to realize what was going on here. It looks fine, but I might recommend renaming the nested `temp_dir` even if it is just something like `temp_dir2`. I know logically it is fine as it is, but it makes it unnecessarily tricky to read. That said, thanks for cleaning up all these tests!